resources/lib/library.py

```diff
@@ -1383,28 +1383,32 @@ def detectPluginContent(self, item):
             # if it's missing it means this is a main directory listing and no need to scan the underlying listitems.
             return None
 
+        if not item.has_key("showtitle") and not item.has_key("artist"):
+            #these properties are only returned in the json response if we're looking at actual file content...
+            # if it's missing it means this is a main directory listing and no need to scan the underlying listitems.
+            return "files"
         if not item.has_key("showtitle") and item.has_key("artist"):
             ##### AUDIO ITEMS ####
-            if item["artist"][0] == item["title"]:
+            if item["type"] == "artist" or item["artist"][0] == item["title"]:
                 return "artists"
-            elif item["album"] == item["title"]:
+            elif item["type"] == "album" or item["album"] == item["title"]:
                 return "albums"
-            elif (item["type"] == "song" or (item["artist"] and item["album"])):
-                return "songs"
+            elif (item["type"] == "song" and not "play_album" in item["file"]) or (item["artist"] and item["album"]):
+                return ("songs", image)
```

Comment:
I'm sure this is correct, but I have to question it (just looking at this code, not the content) - every other instance returns just a string, but here you're returning a tuple?